cmake/CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 # Minimum CMake required
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.18)
```

Comment:
In case anybody is curious, `CheckLinkerFlag` cmake module (used below) is documented in https://cmake.org/cmake/help/latest/module/CheckLinkerFlag.html that:
> New in version 3.18.

Reply:
One of our CI builds is breaking because it's still using CMake 3.10.1. Is there any easy workaround that wouldn't require the `CheckLinkerFlag` module?

Reply:
I just checked and it doesn't appear to be easy to upgrade our CMake version, because we're using a Windows VM maintained by another team. I think we should probably just enable the behavior conditionally, only if `CMAKE_VERSION` is greater than or equal to 3.18 (we already do something like this below).

Reply:
Actually would you mind setting the minimum version to 3.10.1? Let's do that and see if anyone complains. Maybe we can try to gradually increase the minimum version over time so that we can start to rely on newer CMake features.

Reply:
I added fallback code using `check_cxx_source_compiles` function from `CheckCXXSourceCompiles` module.